nova_plugin/keypair.py

```diff
@@ -155,14 +155,23 @@ def validate_path_owner(path):
                   "available on Openstack, but the private key could not be " \
                   "found at {1}".format(ctx.node.properties['resource_id'],
                                         private_key_path)
-            ctx.logger.error('VALIDATION ERROR: ' + err)
+            ctx.logger.error('VALIDATION ERROR: {0}'.format(err))
             raise NonRecoverableError(err)
     else:
         if pk_exists:
             err = 'private key path already exists: {0}'.format(
                 private_key_path)
-            ctx.logger.error('VALIDATION ERROR: ' + err)
+            ctx.logger.error('VALIDATION ERROR: {0}'.format(err))
             raise NonRecoverableError(err)
+        else:
+            err = 'private key directory {0} is not writable'
+            while private_key_path:
+                if os.path.isdir(private_key_path):
+                    if not os.access(private_key_path, os.W_OK):
```

Comment:
if the executable bit is not set for a directory, no reading/writing from it as well.
should test `os.W_OK | os.X_OK`